PGMonitor Service

edenkoveshi opened the pull request:
**Type of Changes:**

 <!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [X] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)



**What is the current behavior? (link to any open issues here)**
Enabling PGMonitor sets up the relevant container, however it can only be scraped using Prometheus kubrenetes pod discovery.


**What is the new behavior (if this is a feature change)?**
Enabling PGMonitor sets up a new service that exposes the relevant port, enabling scraping in any way you'd like.


**Other information**:
The new service name is <cluster-name>-pgmonitor.
The service reconciliation was written very similarly to the existing PGBouncer service.
It is documented and has been tested by me, a few local tests in my minikube environment. Seems OK.

jkatz commented:
I'm OK with the concept overall. The code does require unit tests and a bit more documentation around the function names (following Go standards).

It seems that it was copy and pasted from the pgBouncer bits, as the term "pgBouncer" is in the code comments themselves.

I do think that this should be an opt-in feature, i.e. you should choose if you want to expose the exporter port. That would likely require a CRD change.

edenkoveshi commented:
Added the service feature as optional using a new property in the CRD.
setting `spec.monitoring.pgmonitor.exporter.expose` to `true` will create the service, and it won't be created otherwise.
As I have said, this is very similar to pgBouncer service creation and so comments are quite the same.. changed pgBouncer to pgMonitor though.
